tests/config/test_bless_config.py

```diff
@@ -18,7 +21,61 @@ def test_config_no_password():
     with pytest.raises(ValueError) as e:
         BlessConfig('bogus-region',
                     config_file=os.path.join(os.path.dirname(__file__), 'full.cfg'))
-    assert 'No Region Specific Password Provided.' == e.value.message
+    assert 'No Region Specific And No Default Password Provided.' == e.value.message
+
+    config = BlessConfig('bogus-region',
+                config_file=os.path.join(os.path.dirname(__file__), 'full-with-default.cfg'))
+    assert '<INSERT_DEFAULT_KMS_ENCRYPTED_BASE64_ENCODED_PEM_PASSWORD_HERE>' == config.getpassword()
+
+
+def test_config_environment_override():
+    extra_environment_variables = {
+        'bless_options_certificate_validity_after_seconds': '1',
+        'bless_options_certificate_validity_before_seconds': '1',
+        'bless_options_entropy_minimum_bits': '2',
+        'bless_options_random_seed_bytes': '3',
+        'bless_options_logging_level': 'DEBUG',
+        'bless_options_certificate_extensions': 'permit-X11-forwarding',
+
+        'bless_ca_kms_key_id': 'alias/foo',
+        'bless_ca_us-east-1_password': '<INSERT_US-EAST-1_KMS_ENCRYPTED_BASE64_ENCODED_PEM_PASSWORD_HERE>',
+        'bless_ca_default_password': '<INSERT_DEFAULT_KMS_ENCRYPTED_BASE64_ENCODED_PEM_PASSWORD_HERE>',
+        'bless_ca_ca_private_key_file': '<INSERT_YOUR_ENCRYPTED_PEM_FILE_NAME>',
+        'bless_ca_ca_private_key': base64.b64encode('<INSERT_YOUR_ENCRYPTED_PEM_FILE_CONTENT>'),
+
+        'kms_auth_use_kmsauth': 'True',
+        'kms_auth_kmsauth_key_id': '<INSERT_ARN>',
+        'kms_auth_kmsauth_serviceid': 'bless-test',
+    }
+
+    # Safe the current environment
+    old_environ = os.environ.copy()
```

Comment:
pytest provides a method to monkeypatch the environment variables. Maybe it'd be better to use that and avoid copying/restoring the environment? http://doc.pytest.org/en/latest/monkeypatch.html